### ISSUE TITLE
Davis default configuration doesn't typecheck / nginx config can't be disabled

### ISSUE THREAD

**Issue by TheConfuZzledDude** [NixOS/nixpkgs#317303]
The default value of the `nginx` option at https://github.com/NixOS/nixpkgs/blob/57610d2f8f0937f39dbd72251e9614b1561942d8/nixos/modules/services/web-apps/davis.nix#L223 is `null`, but the type is a `submodule`. This is means that
- It doesn't typecheck without setting `nginx` to some module
- You can't disable its changes to the nginx config easily.

If the intention is to enable it by default, the default value should be set to `{}`, like in the example configuration, and if the intention is to let it be disabled (by default or not), the type should be changed to `nullOr submodule (rest of type)`

**Comment by Ramblurr** [comment]
Thanks for reporting this. The intention is for it to be disabled by default. Are you interested in PRing a fix?

**Comment by TheConfuZzledDude** [comment]
Yep, I'd be happy to, as soon as I can get my own configuration actually working (500 error anywhere except the dashboard and no visible logs anywhere 😅)

**Comment by Elinvention** [comment]
I also have another issue that I can't debug, because I can't see davis's logs. Can we get some logging?

**Comment by sector1209** [comment]
I'm having both the issues mentioned above, abundant 500 errors on any redirect + no logs to be found.

**Comment by sector1209** [comment]
> I'm having both the issues mentioned above, abundant 500 errors on any redirect + no logs to be found.

Oops never mind, my issue was down to an empty string in services.davis.mail.dsn >_< (thanks a million for your config @TheConfuZzledDude )

**Comment by Henry-Hiles** [comment]
It would be nice to have this fixed. I'm also not certain that `mail.dsn` should be required?

**Comment by Ramblurr** [comment]
Hi folks, sorry it took me so long to get to this. 

I've addressed all of the above concerns in https://github.com/NixOS/nixpkgs/pull/409068

- mail dsn is no longer required
- logging is fixed
- nginx virtual host exists by default but can be disabled properly by settings `nginx = null;`

Please give it a test if you can.

Thanks for your patience. Davis is a great piece of software my family and I use every day!

**Comment by Ramblurr** [comment]
Please get in touch if you encounter any more issues or difficulties with davis and the nixos module.
